java-frontend/src/main/java/org/sonar/java/se/constraint/ObjectConstraint.java

```diff
@@ -75,6 +77,17 @@ public boolean hasStatus(@Nullable Object aState) {
     }
     return aState.equals(status);
   }
+  
+  public static ObjectConstraint constraintWithStatus(ProgramState programState, SymbolicValue value, Object aState) {
```

Comment:
In fact I am not sure why this is not a method of ProgramState ? with a dedicated test for it ?